code/modules/mob/living/carbon/human/ai/action_datums/item_pickup.dm

```diff
@@ -28,6 +28,11 @@
 		brain.to_pickup -= to_pickup
 		return ONGOING_ACTION_COMPLETED
 
+	if(brain.primary_weapon && isgun(to_pickup))
```

Comment:
this means it won't be able to pick up guns it could use as a secondary
(not a massive issue tbh, doesn't need to be fixed)